services/dataset.js

```diff
@@ -96,6 +99,37 @@ export const fetchDataset = (id, params = {}) => {
     });
 };
 
+/**
+ * Fetches fields for a specific dataset.
+ *
+ * @param {String} id - dataset id.
+ * @param {String} provider - dataset provider.
+ * @param {String} type - dataset type.
+ * @param {String} provider - dataset table name.
+ * @returns {Object} serialized specified dataset.
+ */
+export const fetchFields = ({ id, provider, type, tableName }) => {
```

Comment:
I think the function is too custom. The service doesn't care how you build the UR:, it receives an URL, fetch it and return data or an error. Keep it simple.

Reply:
@andresgnlez I just moved the code that was present in the "old" service to this one. I'd suggest changing the method `fetchFields` in a separate PR if necessary.